pkg/scheduler/ugm/manager.go

```diff
@@ -631,6 +641,21 @@ func (m *Manager) CanRunApp(queuePath, applicationID string, user security.UserG
 	return userCanRunApp && groupCanRunApp
 }
 
+func (m *Manager) HasWildCardApplied(user string, hierarchy []string) bool {
```

Comment:
This is only used in the clear case but I think we can fold this into the clear itself and only traverse the hierarchy once.
I think that would remove `HasWildCardApplied()`,  `ut.hasWildCardApplied()` and `qt.hasWildCardApplied()`